server/upstream/download-pypi-versions

```diff
@@ -0,0 +1,259 @@
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this library; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301
+# USA
+#
+# (Licensed under the LGPLv2.1 or later)
+#
+#
+# Authors: Thomas Bechtold <tbechtold@suse.com>
+#
+
+import argparse
+import functools
+import os
+import sys
+import time
+from multiprocessing import Pool
+
+try:
+    import xmlrpclib
+except ImportError:
+    # Python3 support
+    import xmlrpc.client as xmlrpclib
+
+
+XMLRPC_SERVER_PROXY = 'https://pypi.python.org/pypi'
+
+
+@functools.total_ordering
+class PackageInfo(object):
+    """Represents a package info"""
+    def __init__(self, type_, name, version, url):
+        self.type = type_.rstrip()
+        self.name = name.rstrip()  # the original name from pypi (without python3- or python- prefix)
+        self.version = version.rstrip()
+        self.url = url.rstrip()
+
+    def __repr__(self):
+        return "PackageInfo obj <%s:%s:%s:%s>" % (self.type, self.name, self.version, self.url)
+
+    def __eq__(self, other):
+        if hasattr(other, 'name'):
+            return self.name == other.name
+
+    def __lt__(self, other):
+        if hasattr(other, 'name'):
+            return self.name < other.name
+
+
+def __get_save_file_serial_name(save_file):
+    """filename to use to store the latest used changelog serial from pypi"""
+    return save_file + "-changelog-serial"
+
+
+def __get_cached_changelog_last_serial(save_file):
+    """try to read the latest changelog serial and return
+    the number or None if not available"""
+    serial_file = __get_save_file_serial_name(save_file)
+    if os.path.exists(serial_file):
+        with open(serial_file, 'r') as f:
+            return int(f.readline())
+    # no changelog serial available
+    return None
+
+
+def __write_cached_changelog_last_serial(save_file, changelog_serial=None):
+    """update the cached changelog serial with the current serial from pypi"""
+    if not changelog_serial:
+        client = xmlrpclib.ServerProxy(XMLRPC_SERVER_PROXY)
+        changelog_serial = str(client.changelog_last_serial())
+
+    serial_file = __get_save_file_serial_name(save_file)
+    with open(serial_file, 'w') as sf:
+        sf.write("%s\n" % changelog_serial)
+
+
+def __read_pypi_package_file(save_file):
+    """read the given pypi file into a list of PackageInfo objs"""
+    packages = list()
+    with open(save_file, "r") as f:
+        for line in f.readlines():
+            if line.startswith('#'):
+                continue
+            pack_info = PackageInfo(*line.split(':', 3))
+            # skip python3-* lines
+            if pack_info.name.startswith('python3-'):
+                continue
+            # remove python- prefix
+            if pack_info.name.startswith('python-'):
+                pack_info.name = pack_info.name.replace('python-', '')
```

Comment:
> This will break python modules named "python-foo" (and packaged as python-python-foo). You want to remove only the initial prefix.

Done in https://github.com/toabctl/osc-plugin-collab/commit/2f370e3bf6fce74afc85f5de1f38382d091f9108